time/src/time.rs

```diff
@@ -29,21 +29,152 @@ pub(crate) enum Padding {
 /// (either positive or negative).
 ///
 /// When comparing two `Time`s, they are assumed to be in the same calendar date.
-#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[derive(Clone, Copy)]
+// The alignment is forced to 8 as an heuristic for performances. This is not
+// required for safety.
+#[repr(C, align(8))]
 pub struct Time {
+    // The order of this struct's fields matter!
+    // Do not change them.
+
+    // Little endian version
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    nanosecond: u32,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    second: u8,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    minute: u8,
+    #[cfg(target_endian = "little")]
     #[allow(clippy::missing_docs_in_private_items)]
     hour: u8,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    padding: Padding,
+
+    // Big endian version
+    #[cfg(target_endian = "big")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    hour: u8,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     minute: u8,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     second: u8,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     nanosecond: u32,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     padding: Padding,
```

Comment:
The placement of this isn't correct for big endian platforms. It would cause the size of the struct to be larger.

Reply:
My bad, fixed that!